Src/WindowsServer/WindowsServer.Shared/Implementation/AzureMetadataRequestor.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Runtime.Serialization.Json;
+    using System.Threading.Tasks;
+#if NETSTANDARD1_3
+    using System.Net.Http;
+#endif
+    using Extensibility;
+    using Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts;
+
+    internal class AzureMetadataRequestor : IAzureMetadataRequestor
+    {
+        /// <summary>
+        /// Azure Instance Metadata Service exists on a single non-routable IP on machines configured
+        /// by the Azure Resource Manager. See <a href="https://go.microsoft.com/fwlink/?linkid=864683">to learn more.</a>
+        /// </summary>
+        internal const string AzureImsApiVersion = "api-version=2017-08-01"; // this version has the format=text capability
+        internal const string AzureImsJsonFormat = "format=json";
+        internal const int AzureImsMaxResponseBufferSize = 512;
+
+        /// <summary>
+        /// Default timeout for the web requests made to obtain Azure IMS data. Internal to expose to tests.
+        /// </summary>
+        internal TimeSpan AzureImsRequestTimeout = TimeSpan.FromSeconds(10);
+
+        /// <summary>
+        /// Private function for mocking out the actual call to IMS in unit tests. Available to internal only.
+        /// </summary>
+        /// parameter sent to the func is a string representing the Uri to request Azure IMS data from.
+        /// <returns>An instance of AzureInstanceComputeMetadata or null.</returns>
+        private Func<string, Task<AzureInstanceComputeMetadata>> azureIMSRequestor = null;
+
+        internal AzureMetadataRequestor(Func<string, Task<AzureInstanceComputeMetadata>> makeAzureIMSRequestor = null)
+        {
+            this.azureIMSRequestor = makeAzureIMSRequestor;
+        }
+
+        /// <summary>
+        /// Gets or sets the base URI for the Azure Instance Metadata service. Internal to allow overriding in test.
+        /// </summary>
+        internal string BaseAimsUri { get; set; } = "http://169.254.169.254/metadata/instance/compute";
+
+        public async Task<AzureInstanceComputeMetadata> GetAzureComputeMetadataAsync()
```

Comment:
I commented on this previously. It's not necessary to use `async` `await` if all you're doing in the body is calling another Task-returning method and returning its result directly. Replace with:
```C#
public Task<AzureInstanceComputeMetadata> GetAzureComputeMetadataAsync()
{
    string metadataRequestUrl = $"{etc...";
    return this.MakeAzureMetadataRequestAsync(metadataRequestUrl);
}
```

Reply:
This seems odd to me, but I think I get it, the `async` part is not necessary to `await` - just that it returns a `Task<>`, thanks for the heads up.